test/box-luatest/downgrade_test.lua

```diff
@@ -783,3 +724,87 @@ g.test_downgrade_user_auth_history_and_last_modified = function(cg)
         check_NO_new_auth_fields()
     end)
 end
+
+-- Check that SQL CHECK constraints could be downgraded.
+g.test_downgrade_sql_tuple_check_contraints = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        local _func = box.space._func
+        local _ck = box.space._ck_constraint
+
+        box.execute([[CREATE TABLE t(i INT PRIMARY KEY, a INT);]])
+        box.execute([[ALTER TABLE t ADD CONSTRAINT c CHECK (a > 10);]])
+
+        local s = box.space._space.index[2]:get{'T'}
+        local f = s.flags.constraint.C
+        t.assert_equals(_func:get{f}.body, 'a > 10')
+
+        local app_version = helper.app_version('2.11.0')
+        t.assert_equals(box.schema.downgrade_issues(app_version), {})
+        box.schema.downgrade(app_version)
+        t.assert_equals(_func:get{f}.body, 'a > 10')
+
+        local prev_version = helper.prev_version('2.11.0')
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {})
+        t.assert_equals(_func:get{f}.body, 'a > 10')
+
+        box.schema.downgrade(prev_version)
+        s = box.space._space.index[2]:get{'T'}
+        t.assert(_func:get{f} == nil)
+        t.assert(s.flags.constraint == nil)
+        t.assert_equals(_ck:get{s.id, 'C'}, {s.id, 'C', false, 'SQL', 'a > 10',
+                        true})
+
+        -- idempotence check
+        box.schema.downgrade(prev_version)
+        s = box.space._space.index[2]:get{'T'}
+        t.assert(_func:get{f} == nil)
+        t.assert(s.flags.constraint == nil)
+        t.assert_equals(_ck:get{s.id, 'C'}, {s.id, 'C', false, 'SQL', 'a > 10',
+                        true})
```

Comment:
This is the exact check as on first downgrade. Let's move into function as in other tests.

Reply:
Done.